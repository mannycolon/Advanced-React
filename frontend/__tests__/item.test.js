import { shallow } from 'enzyme'
import ItemComponent from '../components/Item'
import toJSON from 'enzyme-to-json'

const fakeItem = {
  id: 'abc1234',
  title: 'A Cool Item',
  price: 4000,
  description: 'This Item is really cool!',
  image: 'gog.jpg',
  largeImage: 'largedog.jpg',
}

describe('<Item/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem}/>)
    expect(toJSON(wrapper)).toMatchSnapshot()
  });
  // it('Renders and displays properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem}/>)
  //   const img = wrapper.find('img')
  //   expect(img.props().src).toBe(fakeItem.image)
  //   expect(img.props().alt).toBe(fakeItem.title)
  // });

  // it('Renders out the buttons properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem}/>)
  //   const buttonList = wrapper.find('.buttonList')
  //   expect(buttonList.children()).toHaveLength(3)
  //   expect(buttonList.find('Link')).toHaveLength(1)
  //   expect(buttonList.find('AddToCart').exists()).toBe(true)
  //   expect(buttonList.find('DeleteItem').exists()).toBe(true)
  // });
});
