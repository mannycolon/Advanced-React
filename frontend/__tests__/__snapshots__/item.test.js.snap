// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ForwardRef>
  <img
    alt="A Cool Item"
    src="gog.jpg"
  />
  <ForwardRef>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "abc1234",
          },
        }
      }
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </ForwardRef>
  <ForwardRef>
    $40
  </ForwardRef>
  <p>
    This Item is really cool!
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "abc1234",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="abc1234"
    />
    <DeleteItem
      id="abc1234"
    >
      Delete Item
    </DeleteItem>
  </div>
</ForwardRef>
`;
